lib/view/login_screen.dart

```diff
@@ -0,0 +1,116 @@
+import 'package:flutter/material.dart';
+import 'package:the_quran/controller/authentication_controller.dart';
+import 'package:the_quran/model/authentication_model.dart';
+import 'package:the_quran/view/home_page.dart';
+import 'package:the_quran/view/forgot_password_screen.dart';
+import 'package:the_quran/view/sign_up_screen.dart';
+
+class LoginScreen extends StatelessWidget {
+  final AuthenticationController authController;
+
+  LoginScreen({required this.authController});
+
+  @override
+  Widget build(BuildContext context) {
+    TextEditingController emailController = TextEditingController();
+    TextEditingController passwordController = TextEditingController();
+
+    return Scaffold(
+      appBar: AppBar(
+        title: Text('Log In'),
+      ),
+      body: Padding(
+        padding: EdgeInsets.all(16),
+        child: Column(
+          children: [
+            TextFormField(
+              controller: emailController,
+              decoration: InputDecoration(labelText: 'Email'),
+              validator: (value) {
+                if (value == null || value.isEmpty) {
+                  return 'Please enter your email';
+                }
+                if (!RegExp(r'^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$')
+                    .hasMatch(value)) {
+                  return 'Please enter a valid email';
+                }
+                return null;
+              },
```

Comment:
Do we really need this validator? 

`signInWithEmailAndPassword` method: `FirebaseAuthException` maybe thrown with error code `invalid-email` thrown if the email address is not valid.

Also for the password.

Reply:
No but I thought the task required us to manually validate emails
it if't okay to rely on firebase then I will remove it